handlers.go

```diff
@@ -158,9 +167,70 @@ func (f *faucet) authOAuthUrl(msg *apirest.APIdata, ctx *httprouter.HTTPContext)
 		return ctx.Send(new(HandlerResponse).SetError(ReasonErrOauthProviderNotFound).MustMarshall(), CodeErrOauthProviderNotFound)
 	}
 
-	type urlResponse struct{
+	type urlResponse struct {
 		Url string `json:"url"`
 	}
 	authURL := urlResponse{Url: provider.GetAuthURL(redirectURL)}
 	return ctx.Send(new(HandlerResponse).Set(authURL).MustMarshall(), apirest.HTTPstatusOK)
 }
+
+func (f *faucet) authAragonDaoHandler(msg *apirest.APIdata, ctx *httprouter.HTTPContext) error {
+	var err error
+
+	amount, ok := f.authTypes["aragondao"]
+	if !ok || amount == 0 {
+		return ctx.Send([]byte("auth type AragonDao not supported"), apirest.HTTPstatusInternalErr)
+	}
+
+	type r struct {
+		Data      string         `json:"data"`
+		Signature types.HexBytes `json:"signature"`
+		Network   string         `json:"network"`
+	}
+	newRequest := r{}
+	if err := json.Unmarshal(msg.Data, &newRequest); err != nil {
+		return ctx.Send(new(HandlerResponse).SetError(err.Error()).MustMarshall(), CodeErrIncorrectParams)
+	}
+
+	// Obtains the URL and verifies the signature is from today
+	var addr common.Address
+	if addr, err = aragondaohandler.VerifyAragonDaoRequest(newRequest.Data, newRequest.Signature); err != nil {
+		return ctx.Send(new(HandlerResponse).SetError(err.Error()).MustMarshall(), CodeErrAragonDaoSignature)
+	}
+
+	// Check if the address is already funded
+	if funded, t := f.storage.checkIsFundedAddress(addr); funded {
+		errReason := fmt.Sprintf("address %s already funded, wait until %s", addr.Hex(), t)
+		return ctx.Send(new(HandlerResponse).SetError(errReason).MustMarshall(), CodeErrFlood)
+	}
+
+	// Check if the address is an Aragon DAO address by checking to AragonGraphQL
+	if newRequest.Network != "" {
```

Comment:
This is a bit weird, I'd either force the user to define network or not (and check all configured networks until one is found). Probably the second is easier for the UI side.

Reply:
The reason behind this is offering both possibilities,
1. Searching in just one specific network
2. Searching in any network

In the scaffold, we don't know (yet) the network that should be used (as it's "network-agnostic"). I think this is not the case for the Aragon Plugin.

Anyway, will ask about the logic to @jpaulet